compiler/rustc_span/src/lib.rs

```diff
@@ -782,13 +785,25 @@ impl Span {
     ///     ^^^^^^^^^^^^^^^^^
     /// ```
     pub fn until(self, end: Span) -> Span {
-        let span = self.data();
-        let end = end.data();
+        let span_data = self.data();
```

Comment:
Most of this function's body is copied from `to`. We can't just do `self.to(end.shrink_to_lo())`, because `to` also does some magic where it uses min/max so it can handle overlapping spans (I think, the logic is a bit wonky).

Reply:
Add a comment saying so _or_ refactor the common logic to a new method.